core.js

```diff
@@ -0,0 +1,685 @@
+      
+  //for each route
+  for(routeI = this.routes.length - 1; routeI >= 0; routeI--){
+    route = this.routes[routeI];
+    //establish all connections between this client's publishers and
+    // appropriate subscribers
+    this.connectRoutePubClient(route, pubClient);
+
+    //now check if this client's subscribers match this route
+    subClient = client;
+    
+    //for each publisher that matches this route
+    for(publisherI = route.from.matched.length - 1;
+        publisherI >= 0;
+        publisherI--){
+      pubClient = route.from.matched[publisherI].client;
+      publisher = route.from.matched[publisherI].endpoint;
+      //if this client matches the route's "to"
+      if(route.matchesPubToClient(pubClient, publisher, subClient)){
+        //for each subscriber in this client
+        for(subscriberI = subClient.subscribers.length - 1;
+            subscriberI >= 0;
+            subscriberI--){
+          subscriber = subClient.subscribers[subscriberI];
+          //if this subscriber matches the route/publisher combo
+          if(route.matchesPair(pubClient, publisher,
+                               subClient, subscriber)){
+            //connect the subscriber to this publisher
+            pubClient.addOutConnection(publisher, subClient, subscriber, route);
+            //and add the connection the other way...
+            subClient.addInConnection(pubClient, publisher, subscriber, route);
+          }
+        }
+      }
+    }
+  }
+};
+
+/**
+ * Gets the index of the first client that matches the provided client.
+ * @param client {Object|string|Leaf} the client data to match against
+ * @return {number} The index of the first client in the router's
+ *   client list that matches (name and all metadata info)
+ *   the provided client. Returns -1 if no match is found.
+ */
+Manager.prototype.indexOfClient = function(client){
+  for(var i = 0; i < this.clients.length; i++){
+    if (this.clients[i].matches(client)){
+      return i;
+    }
+  }
+  return -1;
+};
+
+/**
+ * Gets the index of the first admin that matches the provided admin.
+ * @param client {Object|string|Admin} the admin data to match against
+ * @return {number} The index of the first admin in the router's
+ *   admin list that matches the provided admin. 
+ *   Returns -1 if no match is found.
+ */
+Manager.prototype.indexOfAdmin = function(admin){
+  for(var i = 0; i < this.admins.length; i++){
+    if (this.admins[i].matches(admin)){
+      return i;
+    }
+  }
+  return -1;
+};
+
+/**
+ * cleans up both sides of this pub/sub connection
+ * @param {Object} sourceEndpoint The publisher or subscriber that
+ *   the connection is defined in.
+ * @param {Object} sourceConnection The connection associated with the
+ *   sourceEndpoint that we want to clean up.
+ * @throws Throws an error if the connection is not registered with the
+ *   provided endpoint.
+ */
+Manager.breakConnection = function(sourceEndpoint, sourceConnection){
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Function 'anonymous' has a complexity of 7.](https://www.codacy.com/app/quinkennedy/spacebrew/file/2279533613/issues/source?bid=3184377&fileBranchId=3187330#l232)